Forge.TreeWalker/src/ForgeSchemaValidator.cs

```diff
@@ -0,0 +1,263 @@
+                    schemaList.Add(SerializeToJObject(item));
+                }
+            }
+
+            return schemaList;
+        }
+
+        private static List<JObject> ConvertStringToJObjectList(string schema, bool validateAsDictionary)
+        {
+            List<JObject> schemaList = new List<JObject>();
+
+            //There could be three possible cases:
+            //1. TreeName mapped to the ForgeTree in the dictionary and custom rules handle dictionary.
+            //2. There are only ForgeTree without matching forge tree name custom rules handle ForgeTree list.
+            //3. The string schema that could be deserialized as a ForgeTree should be parsed directly
+            Dictionary<string, ForgeTree> forgeTrees = JsonConvert.DeserializeObject<Dictionary<string, ForgeTree>>(schema);
+
+            if (validateAsDictionary)
+            {
+                schemaList.Add(JObject.Parse(schema));
+            }
+            else
+            {
+                foreach (KeyValuePair<string, ForgeTree> kvp in forgeTrees)
+                {
+                    ForgeTree forgeTree = kvp.Value;
+
+                    if (forgeTree.Tree == null)
+                    {
+                        // Deserialize into Dictionary does not throw exception but will have null "Tree" property if schema is just a ForgeTree.
+                        // try to deserialize string to forge tree directly
+                        JsonConvert.DeserializeObject<ForgeTree>(schema);
+                        schemaList.Add(JObject.Parse(schema));
+                        break;
+                    }
+
+                    schemaList.Add(SerializeToJObject(forgeTree));
+                }
+            }
+
+            return schemaList;
+        }
+
+        private static JObject SerializeToJObject(object forgeTree)
+        {
+            string stringSchema = JsonConvert.SerializeObject(
+                forgeTree,
+                new JsonSerializerSettings
+                {
+                    DefaultValueHandling = DefaultValueHandling.Ignore, // Prevent default values from getting added to serialized json schema.
+                    Converters = new List<JsonConverter> { new Newtonsoft.Json.Converters.StringEnumConverter() } // Use string enum values instead of numerical.
+                });
+
+            return JObject.Parse(stringSchema);
+        }
+
+        private static List<JObject> GetSchemaFromPath(string path, bool validateAsDictionary)
+        {
+            string schema = File.ReadAllText(path);
+            List<JObject> res = ConvertStringToJObjectList(schema, validateAsDictionary);
+
+            return res;
+        }
+
+        private static List<JObject> GetAllSchemasInDirectory(string path, bool validateAsDictionary)
+        {
+            List<JObject> schemaList = new List<JObject>();
+            string[] files = Directory.GetFiles(path);
+            List<object> schemalist = new List<object>();
```

Comment:
Remove this line. Looks like you left this here from some earlier testing or implementation perhaps.